model/adjuster/anonymizer.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package adjuster
+
+import "github.com/uber/jaeger/model"
+
+// Anonymizer returns an adjuster that modifies the trace to remove any information that might
+// be considered private or confidential. Examples are service names, port numbers, operations,
+// and any user-added data.
+//
+// This allows real production traces to safely be kept in public repositories for use in test
+// suites, after anonymization.
+//
+// A custom mapping is provided that dictates how to anonymize data. If a value is encountered that
+// that not in the mapping, the mapping is modified to include the new value. Modifications are
+// deterministic.
+//
+// This adjuster never returns any errors.
+func Anonymizer(mapping AnonymizerMapping) Adjuster {
+	return Func(func(trace *model.Trace) (*model.Trace, error) {
+		return trace, nil
+	})
+}
+
+// AnonymizerMapping dictates how anonymization is done.
+//
+// It is a map from anonymized field name -> field value -> anonymized value
+//
+// An example of a mapping:
+//var mapping = AnonymizerMapping{
+//	adjuster.AnonymizedService: {
```

Comment:
I'm ok with having an Anonymizer struct.

The reason why I had it as an explicit mapping is so that it can be consistent across multiple traces, meaning if trace1 has services `foo` and `bar` and trace2 has services `bar`, then `bar` would be anonymized to `service2` in both traces. Thinking about it now, I'm not sure if that really matters or not from the programmer's point of view. It certainly does reduce the effectiveness of anonymization, though, so maybe it's not such a great idea.

I do agree that there will need to be a standalone binary for convenience.